### DISCUSSION TITLE
RenovateBot to track generic archive packages

### DISCUSSION THREAD

**Discussion by deep1993nov** [renovatebot/renovate#22870] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.131.0

### If you're self-hosting Renovate, select which platform you are using.

GitHub Enterprise Server

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

We are developing terraform modules and pushing to artifactory as a zip file.
We want to track it using renovatebot in such a way that it will go over the artifactory and look for updated versions of a zip.
We are using schematic versioning. The name of package is : NAME_SCHEMVERSION.tar.gz

We are using regex manager to track it.

### What you tried so far.

We are using regex manager to track the dependecy. We are confused about which data source to use for.

{  
  "extends": ["config:base"],  
  "customManagers": [  
    {  
      "fileMatch": ["^path/to/your/terraform/module\\.tf$"],  
      "matchStrings": [  
        "source\\s*=\\s*\"https://artifacts\\.com/artifactory/iac-terraform/releases/(?<depName>[-_a-zA-Z0-9]+)/(?<currentValue>[-_a-zA-Z0-9.]+)\\.tgz\""  
      ],  
      "datasourceTemplate": "url",  
      "versioningTemplate": "semver",  
      "registryUrls": [  
        "https://artifacts.com/artifactory/iac-terraform//terraform/releases/"  
      ]  
    }  
  ]  
}

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by deep1993nov** [comment]
@rarkins 
Could you please tell whether this is something Renovatebot supports
If required, I can provide more info about this

**Reply by rarkins** [reply, marked as answer]
Not supported. PRs welcome for a new datasource to support

**Comment by oemel09** [comment]
AFAICT this is supported by the `artifactory` datasource

**Reply by HonkingGoose** [reply]
Are you sure about that `artifactory` datasource bit? It looks like `artifactory` is for _Conan_ packages. [^1]

The user's configuration mentions `.tf` files, those are for Terraform. Renovate now has a Terraform package manager. [^2]

[^1]: [Renovate docs, `artifactory` datasource](https://docs.renovatebot.com/modules/datasource/artifactory/)
[^2]: [Renovate docs, Terraform manager](https://docs.renovatebot.com/modules/manager/terraform/)

**Reply by oemel09** [reply]
I can now confirm that `customManagers` with `customType: regex` do work for `generic` artifactory packages with the `artifactory` datasource. I've no idea about _Conan_ packages, but it worked for my case where I just wanted to update an arbitrary zip file with an arbitrary versioning.

My case this is also independent from terraform, I just happened to land in this discussion because of the generic artifactory package.
